tests/test_config.py

```diff
@@ -13,34 +13,35 @@
 class TestConfig:
     @pytest.fixture(autouse=True)
     def _init_tmpdir(self, tmpdir):
+        os.environ["HOME"] = str(tmpdir)
```

Comment:
This change will leak to other tests. You have to use a fixture that resets the original value. You also want to drop any `XDG_*` env vars to force default values.

Something like this should work (untested).

```
# add to conftest.py
import os
from unittest import mock

@pytest.fixture(autouse=True)
def tmp_home(tmpdir):
    with mock.patch.dict(os.environ):
        os.environ["HOME"] = str(tmpdir)
        for key in list(os.environ):
        if key.startswith("XDG_"):
            os.environ.pop(key)
       yield
```

Then make it an auto-use fixture.

Reply:
https://github.com/instructlab/instructlab/pull/1617 provides infrastructure to solve this problem.

Reply:
The current tests work well-enough, we should prioritize this before merging that one.

Reply:
You are leaking information between tests case. This will break other cases in very hard to detect ways.

Reply:
PR #1617 provides the infrastructure to fix the problem:
- accept `tmp_path_home` instead of `tmpdir`
- drop ` os.environ["HOME"] = str(tmpdir)`